packages/launcher/src/components/launch/Launch.tsx

```diff
@@ -84,24 +92,44 @@ export default function Launch(): ReactElement {
     ipcRenderer.send("request-tags");
     ipcRenderer.send("request-version");
   };
+
+  const getInternetConnection = async () => {
+    const ipcRenderer: IpcRenderer = window.require("electron").ipcRenderer;
+    await ipcRenderer.on("internet-status", (event, arg: boolean) => {
+      setInternetConnection(arg);
+      setCheckedInternetConnection(true);
+    });
+    ipcRenderer.send("request-internet");
+  };
+
   if (!requested) {
     getVersions();
     setRequested(true);
     return <CircularProgress />;
   }
 
+  if (!checkedInternetConnection) {
+    getInternetConnection();
+    return <CircularProgress className={classes.progress} />;
+  }
+
   const handleClose = () => {
     setClicked(true);
   };
 
   const startContainer = async () => {
     const connected = await docker.isConnected();
     if (connected) {
-      // setPulling(true);
-      setStatus("Pulling Axon image");
-      await docker.pullImage(axonVersion);
-      // setPulling(false);
-      setStatus("Finished pulling.");
+      if (internetConnection) {
+        console.log("Valid Internet Connection");
+        setStatus("Pulling Axon image");
+        await docker.pullImage(axonVersion);
+        // setPulling(false);
+        setStatus("Finished pulling.");
+      } else {
+        console.log("No Internet Connection Detected, Skipping Pulling Images");
```

Comment:
What happens if the containers do not exist already? Is it possible for this check to fail because of the if statement on line 111?

Reply:
need to see if at least axon:AXON_VERSION exists here. in the actual server, need to see if the respective images exist.